src/ReadSpeed.hxx

```diff
@@ -52,35 +59,88 @@ struct EntryRange {
    Long64_t fEnd = -1;
 };
 
+struct ByteData {
+   ULong64_t fUncompressedBytesRead;
+   ULong64_t fCompressedBytesRead;
+};
+
+std::vector<std::string> GetMatchingBranchNames(const std::string &fileName, const std::string &treeName,
+                                                const std::vector<std::string> &regexes)
+{
+   TFile *f = TFile::Open(fileName.c_str());
+   if (f->IsZombie())
+      throw std::runtime_error("Could not open file '" + fileName + '\'');
+   std::unique_ptr<TTree> t(f->Get<TTree>(treeName.c_str()));
+   if (t == nullptr)
+      throw std::runtime_error("Could not retrieve tree '" + treeName + "' from file '" + fileName + '\'');
+
+   const auto unfilteredBranchNames = ROOT::Internal::RDF::GetTopLevelBranchNames(*t);
+   std::set<std::string> usedRegexes;
+   std::vector<std::string> branchNames;
+
+   auto filterBranchName = [regexes, &usedRegexes](std::string bName) {
+      const auto matchBranch = [&usedRegexes, bName](std::string regex) {
+         std::regex branchRegex(regex);
+         bool match = std::regex_match(bName, branchRegex);
+
+         if (match)
+            usedRegexes.insert(regex);
+
+         return match;
+      };
+
+      const auto iterator = std::find_if(regexes.begin(), regexes.end(), matchBranch);
+      return iterator != regexes.end();
+   };
+   std::copy_if(unfilteredBranchNames.begin(), unfilteredBranchNames.end(), std::back_inserter(branchNames),
+                filterBranchName);
+
+   if (branchNames.empty())
+      throw std::runtime_error("Provided branch regexes didn't match any branches in the tree.");
+   if (usedRegexes.size() != regexes.size()) {
+      std::stringstream errStream;
```

Comment:
Ah TIL, in Java it's always recommended to use StringBuilder instead of regular appending for performance reasons, interesting that it's the opposite in CPP